src/frustum_models/three_dimensional_lidar_frustum.cpp

```diff
@@ -0,0 +1,138 @@
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Simbe Robotics, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Steve Macenski (steven.macenski@simberobotics.com)
+ *********************************************************************/
+
+#include <spatio_temporal_voxel_layer/frustum_models/three_dimensional_lidar_frustum.hpp>
+
+namespace geometry
+{
+
+/*****************************************************************************/
+ThreeDimensionalLidarFrustum::ThreeDimensionalLidarFrustum(const double &vFOV, const double &hFOV,
+                                                           const double &min_dist, const double &max_dist) : _vFOV(vFOV), _hFOV(hFOV), _min_d(min_dist), _max_d(max_dist)
+/*****************************************************************************/
+{
+  _valid_frustum = true;
+  ros::NodeHandle nh;
+
+  _vFOVhalf = _vFOV / 2.0;
+  _hFOVhalf = _hFOV / 2.0;
+}
+
+/*****************************************************************************/
+ThreeDimensionalLidarFrustum::~ThreeDimensionalLidarFrustum(void)
+/*****************************************************************************/
+{
+}
+
+/*****************************************************************************/
+void ThreeDimensionalLidarFrustum::TransformModel(void)
+/*****************************************************************************/
+{
+}
+
+/*****************************************************************************/
+bool ThreeDimensionalLidarFrustum::IsInside(const openvdb::Vec3d &pt)
+/*****************************************************************************/
+{
+
+  Eigen::Vector3d point_in_global_frame(pt[0], pt[1], pt[2]);
+  Eigen::Vector3d transformed_point = _orientation.conjugate() * (point_in_global_frame - _position);
+
+  double radial_distance = sqrt((transformed_point[0] * transformed_point[0]) + (transformed_point[1] * transformed_point[1]));
```

Comment:
Oh sorry, I can be more clear, you calculate a distance `radial_distance` which instead what you could do is calculate `radial_distance_squared` (r*r). What this lets you do then is skip the `sqrt()` operation that's pretty heavy, and then rather than comparing `radial_distance` with `max_distance` or `min_distance` you can compare it to `max_distance_squared` and `min_distance_squared`. You can store those distance squared variables as class variables on initialization in the constructor. 

This lets you skip number_of_points * sqrt_cost CPU time for no degradation of behaviors

Reply:
Generally my philosophy is to avoid sqrt() and log() whenever possible when you dont need to true distance itself but comparing it to another distance - the squared distance is the same

Reply:
After that, I can merge this. Thanks for the great work here! It's going to be a big addition to the package and your application hopefully!

Reply:
Removed the square root...